src/defu.ts

```diff
@@ -55,9 +56,17 @@ function extend (merger?: Merger): DefuFn {
 const defu = extend() as Defu
 
 // Custom version with function merge support
-defu.fn = extend((obj, key, value) => {
-  if (typeof value === 'function') {
-    obj[key] = value(obj[key])
+defu.fn = extend((obj, key, currentValue) => {
+  if (typeof obj[key] !== 'undefined' && typeof currentValue === 'function') {
```

Comment:
Supporting `undefined`, allows users giving a function to be executed on merge (like defaults `port: undefined`). What is the failing case?

Reply:
Well i think if the value is not defined in the defaults, it makes no sense to call the function with an undefined value as parameter since it can directly provide the value 🤔

Reply:
There is a different:

```js
const obj = { foo: undefined } // 'foo' in obj === true
```

So will be only executed if defaults explicitly defined key but no value.

Reply:
Then we would need to change this test: https://github.com/nuxt-contrib/defu/pull/21/files#diff-85398cecb8bcdcabad31a279f8664f3aR79-R95

Reply:
ok let's keep it as is :)